### PR TITLE
MMF-233 Drop Manual Rules and Issues features

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java [hunk 1]
```diff
@@ -144,10 +144,14 @@ public String resolution() {
     return getNullableField(IssueIndexDefinition.FIELD_ISSUE_RESOLUTION);
   }
 
+  /**
+   * @deprecated since 5.5
+   */
+  @Deprecated
   @Override
   @CheckForNull
   public String reporter() {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueDoc.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-web/src/main/js/components/source-viewer/popups/line-actions-popup.js [hunk 2]
```diff
@@ -26,7 +25,6 @@ export default Popup.extend({
 
   events: {
     'click .js-get-permalink': 'getPermalink',
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/components/source-viewer/popups/line-actions-popup.js]: BLOCKER Remove this trailing comma. [rule](https://dory.sonarsource.com/coding_rules#rule_key=javascript%3ATrailingComma)

File: sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java [hunk 3]
```diff
@@ -48,6 +48,10 @@
   public static final String ON_COMPONENT_ONLY = "onComponentOnly";
   public static final String RULES = "rules";
   public static final String ACTION_PLANS = "actionPlans";
+  /**
+   * @deprecated since 5.5, manual issue feature has been dropped.
+   */
+  @Deprecated
   public static final String REPORTERS = "reporters";
```
Comment (SonarTech) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java [hunk 4]
```diff
@@ -341,15 +340,14 @@ public DefaultIssue setResolution(@Nullable String s) {
     return this;
   }
 
+  /**
+   * @deprecated since 5.5, manual issue feature has been dropped.
+   */
+  @Deprecated
   @Override
   @CheckForNull
   public String reporter() {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/issue/DefaultIssue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-scanner-engine/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java [hunk 5]
```diff
@@ -92,9 +93,14 @@ public String resolution() {
     return issue.resolution();
   }
 
+  /**
+   * @deprecated since 5.5, manual issue feature has been dropped.
+   */
+  @Deprecated
+  @CheckForNull
   @Override
   public String reporter() {
```
Comment (SonarTech) [inline sonar-scanner-engine/src/main/java/org/sonar/batch/issue/TrackedIssueAdapter.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java [hunk 6]
```diff
@@ -139,8 +139,9 @@
   String resolution();
 
   /**
-   * Login of the user who reported this issue. Null if the issue is reported by a rule engine.
+   * @deprecated since 5.5, manual issue feature has been dropped.
    */
+  @Deprecated
   @CheckForNull
   String reporter();
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/rule/RuleKey.java [hunk 7]
```diff
@@ -33,7 +33,12 @@
 @Immutable
 public class RuleKey implements Serializable, Comparable<RuleKey> {
 
+  /**
+   * @deprecated since 5.5, manual rule feature has been dropped
+   */
+  @Deprecated
   public static final String MANUAL_REPOSITORY_KEY = "manual";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/rule/RuleKey.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/rule/RuleKey.java [hunk 8]
```diff
@@ -77,8 +82,12 @@ public String rule() {
     return rule;
   }
 
+  /**
+   * @deprecated since 5.5, manual rule feature has been dropped
+   */
+  @Deprecated
   public boolean isManual() {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/rule/RuleKey.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-core/src/main/java/org/sonar/core/issue/DefaultIssueBuilder.java [hunk 9]
```diff
@@ -114,9 +113,12 @@ public DefaultIssueBuilder effortToFix(@Nullable Double d) {
     return this;
   }
 
+  /**
+   * @deprecated since 5.5, manual issue feature has been dropped.
+   */
+  @Deprecated
   @Override
   public DefaultIssueBuilder reporter(@Nullable String s) {
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/issue/DefaultIssueBuilder.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/issue/Issuable.java [hunk 10]
```diff
@@ -112,8 +113,10 @@
     IssueBuilder severity(@Nullable String severity);
 
     /**
-     * Login of the user who reported the issue. Optional.
+     * @deprecated since 5.5, manual issue feature has been dropped.
      */
+    @Deprecated
+    @CheckForNull
     IssueBuilder reporter(@Nullable String reporter);
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/issue/Issuable.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 14 issues:
* BLOCKER 1 blocker
* MAJOR 3 major
* MINOR 1 minor
* INFO 9 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Private method 'findSourceLineUser' is never used.](https://github.com/SonarSource/sonarqube/blob/f0aa8357017cb0ccbf4fd48456ed3f4fee4ce98c/server/sonar-server/src/main/java/org/sonar/server/issue/IssueService.java#L320) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/f0aa8357017cb0ccbf4fd48456ed3f4fee4ce98c/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L122) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MAJOR [Remove this useless assignment to local variable "that"](https://github.com/SonarSource/sonarqube/blob/f0aa8357017cb0ccbf4fd48456ed3f4fee4ce98c/server/sonar-web/src/main/js/components/source-viewer/main.js#L504) [rule](https://dory.sonarsource.com/coding_rules#rule_key=javascript%3AS1854)
* MAJOR [Remove the declaration of the unused 'that' variable.](https://github.com/SonarSource/sonarqube/blob/f0aa8357017cb0ccbf4fd48456ed3f4fee4ce98c/server/sonar-web/src/main/js/components/source-viewer/main.js#L504) [rule](https://dory.sonarsource.com/coding_rules#rule_key=javascript%3AUnusedVariable)

**Comment by SonarTech** [comment]
SonarQube analysis reported 12 issues:
* MAJOR 2 major
* MINOR 1 minor
* INFO 9 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove this unused "userIndex" private field.](https://github.com/SonarSource/sonarqube/blob/dd915ee2e331cca6add6a102b5b6a9945f630dc0/server/sonar-server/src/main/java/org/sonar/server/issue/IssueService.java#L71) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1068)
* MAJOR [Remove this unused "sourceService" private field.](https://github.com/SonarSource/sonarqube/blob/dd915ee2e331cca6add6a102b5b6a9945f630dc0/server/sonar-server/src/main/java/org/sonar/server/issue/IssueService.java#L72) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1068)
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/dd915ee2e331cca6add6a102b5b6a9945f630dc0/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L122) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues:
* MINOR 1 minor
* INFO 9 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/cd5ac0a8d6c9cf4e663998951b7f7898dae00e3c/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L122) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by julienlancelot** [comment]
@stas-vilchik Could you please review https://github.com/SonarSource/sonarqube/pull/872/commits/7b6c07670f2ef669d1f7a0c78d91d53934660107, https://github.com/SonarSource/sonarqube/pull/872/commits/5aa9b756ee1d649204862b16e5fa819fa63da0fe and https://github.com/SonarSource/sonarqube/pull/872/commits/a94ef49f76e39ba90e1b22f25dea3185cce0b3a0 ? 
Thanks.

File: server/sonar-web/src/main/js/components/navigator/filters/ajax-select-filters.js [hunk 11]
```diff
@@ -453,38 +453,6 @@ const AssigneeFilterView = AjaxSelectFilterView.extend({
 });
 
 
-const ReporterFilterView = AjaxSelectFilterView.extend({
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/components/navigator/filters/ajax-select-filters.js]: This code doesn't relate to the ticket.

Comment (julienlancelot) [inline server/sonar-web/src/main/js/components/navigator/filters/ajax-select-filters.js]: Your right, I'll move it to the removal of the reporter field

**Comment by stas-vilchik** [comment]
@julienlancelot I did some clean up, then LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues:
* MINOR 1 minor
* INFO 9 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/7ec2b9a98d78a7954707f1d94352444c39564bf9/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L122) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by julienlancelot** [comment]
thanks @stas-vilchik !

File: server/sonar-web/src/main/js/components/navigator/filters/ajax-select-filters.js (hunk 11 above)
Comment (julienlancelot) [inline server/sonar-web/src/main/js/components/navigator/filters/ajax-select-filters.js]: Hum, no, in fact this commit already belongs to "SONAR-7473 Drop 'reporter' field from UI" commit.
@stas-vilchik So it should be ok ?

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues:
* MINOR 1 minor
* INFO 9 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/3a8dd1c3951120ce168ca49403fd9875ff333f8e/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L122) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 10 issues:
* MINOR 1 minor
* INFO 9 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/d5c48a3db2b74a08415211a7096aefe50e466a30/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L122) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java [hunk 12]
```diff
@@ -53,6 +53,11 @@
    */
   @Deprecated
   public static final String DEPRECATED_ACTION_PLANS = "actionPlans";
```
Comment (SonarTech) [inline sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 22 issues:
* MAJOR 1 major
* MINOR 9 minor
* INFO 12 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/sonar-plugin-api/src/main/java/org/sonar/api/issue/Issue.java#L185) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
* MINOR [Remove this use of "DEPRECATED_ACTION_PLANS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L117) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java#L122) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MAJOR [This method has 102 lines, which is greater than the 100 lines authorized. Split it into smaller methods.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java#L120) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS138)
* MINOR [Remove this use of "DEPRECATED_ACTION_PLANS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java#L178) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Remove this use of "PLANNED"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java#L182) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Remove this use of "DEPRECATED_ACTION_PLANS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchAction.java#L370) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* INFO [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java#L73) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)
* MINOR [Remove this use of "DEPRECATED_ACTION_PLANS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java#L100) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* MINOR [Remove this use of "REPORTERS"; it is deprecated.](https://github.com/SonarSource/sonarqube/blob/140fa7fbcc5f962ae5cbfa177b257c8114435bfe/sonar-ws/src/main/java/org/sonarqube/ws/client/issue/IssueFilterParameters.java#L100) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* ... 2 more

**Comment by julienlancelot** [comment]
@teryk Please review this commit

**Comment by stas-vilchik** [comment]
@julienlancelot `ManualRulesTest` should be dropped

**Comment by teryk** [comment]
LGTM! List of modifications done on my side:
- api/rules/delete description updated to remove any mention of manual rule
- remove ManualRule IT
- api/rules/update description updated to remove any mention of manual rule
- api/issues/create dropped. I'll do it in a separate commit so you can revert. We should drop this WS, the same way I will drop api/issues/do_action. This WS should not exist anymore as the related feature is dropped

**Comment by teryk** [comment]
I keep the PR so you can check what has been done @julienlancelot

**Comment by julienlancelot** [comment]
Perfect, thanks @teryk !

**Pull request merged**